### ISSUE TITLE
Spinner appears on every paste after initial copy, sometimes not completing

### ISSUE THREAD

**Issue by JakeMarsh-at** [microsoft/vscode#236625]
Type: <b>Bug</b>

> Issue troubleshooting has identified that the issue is with Visual Studio Code. 

- Copy text from editor
- Try to paste that text back in the same file or another file

Expected: text is pasted
Actual: spinner appears, text usually never gets pasted

Can sometimes get past the issue by taking other actions before trying to paste.

VS Code version: Code 1.96.0 (Universal) (138f619c86f1199955d53b4166bef66ef252935c, 2024-12-11T02:29:09.626Z)
OS version: Darwin arm64 23.6.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Max (10 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|24, 17, 12|
|Memory (System)|64.00GB (5.83GB free)|
|Process Argv|--crash-reporter-id a6648a3f-85df-458f-a45c-8947c29596ff|
|Screen Reader|no|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythonnoceb:30805159
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
dvdeprecation:31068756
dwnewjupytercf:31046870
2f103344:31071589
nativerepl2:31139839
pythonrstrctxt:31112756
nativeloc2:31192216
cf971741:31144450
iacca1:31171482
notype1:31157159
5fd0e150:31155592
dwcopilot:31170013
stablechunks:31184530
6074i472:31201624

```

</details>

**Comment by JakeMarsh-at** [comment]
Confirmed the issue still remains after upgrading to 1.96.1.

**Comment by maurodesouza** [comment]
Same issue here 😕

**Comment by mjbvz** [comment]
Can you please share a project  that causes this problem so we can take a look?

Until then the workaround is to set `"typescript.updateImportsOnPaste.enabled": false` / `"javascript.updateImportsOnPaste.enabled": false`

**Comment by maurodesouza** [comment]
Sorry, I cannot share It because it is private, but the workaround worked for me. Thanks!

**Comment by gl-aagostino** [comment]
I have the same problem, but disabling those updates on import worked. Are those new settings? For large projects, I think it's just too much, even on a new PC with a high end CPU and tons of RAM. This only started happening when I updated VS code a two days ago.

**Comment by JakeMarsh-at** [comment]
Can confirm disabling that setting also solved the issue for me — thanks. Unfortunately I cannot share the project that exhibited this behavior, but it is a very large one.

**Comment by jove4015** [comment]
This workaround worked for my project as well!

**Comment by magick93** [comment]
Also happening. Just updated. 

Using 1.96.2  fabdb6a30b49f79a7aba0f2ad9df9b399473380f x64

**Comment by Lakhankargathara** [comment]
I have been experiencing the same difficulty since last month.

https://github.com/user-attachments/assets/a943a17f-07ca-4252-b35c-c2763bf12ad4

**Comment by jasonrundell** [comment]
I can temporarily fix this issue by disabling the Copilot extensions (Pilot and Chat) but the issue comes back only after a few minutes of working on the project again.

**Comment by madmacc** [comment]
I find if I copy from VSCode then paste into notepad and then copy from notepad and back into VScode there is no spinner. 
Not sure if this is helpful.
Incredibly annoying.

**Comment by MylesWardell** [comment]
I have turned off format on paste unsure why this is happening is annoying though. Any short term settings fixes?

**Comment by subroto-biswas-deel** [comment]
Don't think it's a VSCode issue. I disabled all extensions and reloaded the window and the loading stopped.
Anyway, it would be helpful if vscode could display some msg on which extension is taking time automatically in such scenarios. I can then disable it, instead of finding it out myself.

**Comment by andreibtc84** [comment]
Having the same issue. 
Getting this sometimes:
'Contributed command 'editor.action.clipboardPasteAction' does not exist.'

**Comment by madmacc** [comment]
Anyone figured out what is causing this yet?
It is insanely annoying having to copy to notepad and back to VSCode.

**Comment by Richard-Collins** [comment]
I've been getting this too,
1.97.2
Electron 32.2.7

Only Microsoft can fail to make an editor that can reliably cut'n'paste. Impressive guys.

**Comment by spurdow** [comment]
This issue should be reopened because it is still happening.
VSc version: 1.97.2
Apple M2 Pro

**Comment by rishabhpineswift** [comment]
same with my M4 Chip its so annoying

**Comment by brotzky** [comment]
```
macOS 12.5.1 (21G83)

Version: 1.97.2 (Universal)
Commit: e54c774e0add60467559eb0d1e229c6452cf8447
Date: 2025-02-12T23:20:35.343Z
Electron: 32.2.7
ElectronBuildId: 10982180
Chromium: 128.0.6613.186
Node.js: 20.18.1
V8: 12.8.374.38-electron.0
OS: Darwin arm64 21.6.0
```

I'm on latest and this continues to happen. Extremely frustrating bug making the editor feel unusable.

**Comment by gregg-cbs** [comment]
> Can you please share a project that causes this problem so we can take a look?
> 
> Until then the workaround is to set `"typescript.updateImportsOnPaste.enabled": false` / `"javascript.updateImportsOnPaste.enabled": false`

This works. It seems as if vscode or something else has changed this to true without me knowing about it. Made every copy and paste so slow. Thanks for the answer.

**Comment by brotzky** [comment]
> ```
> macOS 12.5.1 (21G83)
> 
> Version: 1.97.2 (Universal)
> Commit: e54c774e0add60467559eb0d1e229c6452cf8447
> Date: 2025-02-12T23:20:35.343Z
> Electron: 32.2.7
> ElectronBuildId: 10982180
> Chromium: 128.0.6613.186
> Node.js: 20.18.1
> V8: 12.8.374.38-electron.0
> OS: Darwin arm64 21.6.0
> ```
> 
> I'm on latest and this continues to happen. Extremely frustrating bug making the editor feel unusable.

ended up just using cursor 😅. would recommend.

**Comment by fsdlead1** [comment]
> Type: Bug
> 
> > Issue troubleshooting has identified that the issue is with Visual Studio Code.
> 
> * Copy text from editor
> * Try to paste that text back in the same file or another file
> 
> Expected: text is pasted Actual: spinner appears, text usually never gets pasted
> 
> Can sometimes get past the issue by taking other actions before trying to paste.
> 
> VS Code version: Code 1.96.0 (Universal) ([138f619](https://github.com/microsoft/vscode/commit/138f619c86f1199955d53b4166bef66ef252935c), 2024-12-11T02:29:09.626Z) OS version: Darwin arm64 23.6.0 Modes:
> 
> System Info
> Extensions: none
> A/B Experiments

I had the same issue, but while loading, just hover over the loading symbol, and you will see an "X" mark. Click the "X," and the content will be pasted.

**Comment by QuentinDstl** [comment]
Can't use vscode everything is slow just for a ctrl c ctrl v, unusable

**Comment by mjbvz** [comment]
Locking this to make https://github.com/microsoft/vscode/issues/235959#issuecomment-2644217357 more obvious

If you're still seeing issues on VS Code 1.98+ with extensions disabled, please open a new issue and we'll take a look
